LeanMapperQuery/Entity.php

```diff
@@ -52,19 +52,23 @@ public static function queryEntityProperty(LeanMapper\Entity $entity, $field, IQ
 		$mapper = $entity->mapper;
 
 		$relationship = $property->getRelationship();
-		if ($relationship instanceof Relationship\BelongsToMany) {
-			$filters[] = function (Fluent $fluent) use ($mapper, $query) {
-				$query->applyQuery($fluent, $mapper);
-			};
+		$strategy = NULL;
 
-			$targetTable = $relationship->getTargetTable();
-			$referencingColumn = $relationship->getColumnReferencingSourceTable();
+		if ($relationship instanceof Relationship\BelongsToMany || $relationship instanceof Relationship\HasMany) {
```

Comment:
Potřebujeme tuhle podmínku. Nemůžeme prostě `$strategy` nastavit vždy?

Reply:
Tahle podmínka je pro případ, kdy si uživatel z nějakého důvodu (třeba kvůli vlastnímu filtru) explicitně u vazby zapne UNION strategii, tak abychom mu to nepřepsali zpátky na IN strategii.

Reply:
Ještě přemýšlím nad touhle podmínkou. Nestačilo by toto:

```suggestion
		$strategy = $relationship->getStrategy();
		if ($strategy !== Result::STRATEGY_UNION) {
			$strategy = $query->getStrategy();
		}
```

Reply:
Ještě přemýšlím nad touto podmínkou. Nestačilo by:

```suggestion
 		$strategy = $relationship->getStrategy();
 		if ($strategy !== Result::STRATEGY_UNION) {
 			$strategy = $query->getStrategy();
 		}
```